cmd/boulder-ca/main.go

```diff
@@ -203,8 +218,34 @@ func main() {
 		issuers = append(issuers, issuer)
 	}
 
-	profile, err := issuance.NewProfile(c.CA.Issuance.Profile, c.CA.Issuance.IgnoredLints)
-	cmd.FailOnError(err, "Couldn't load issuance profile")
+	if c.CA.Issuance.DefaultCertificateProfileName == "" {
+		c.CA.Issuance.DefaultCertificateProfileName = "defaultBoulderCertificateProfile"
+	}
+	logger.Infof("Configured default certificate profile name set to: %s", c.CA.Issuance.DefaultCertificateProfileName)
+
+	deprecatedProfileExists := reflect.ValueOf(c.CA.Issuance.Profile).IsZero()
+	if !deprecatedProfileExists && len(c.CA.Issuance.CertProfiles) > 0 {
+		cmd.Fail("Only one of Issuance.Profile or Issuance.CertProfiles can be configured")
+	}
+
+	certProfilesMap := make(map[string]issuance.ProfileConfig, 0)
+	if !deprecatedProfileExists {
+		certProfilesMap[c.CA.Issuance.DefaultCertificateProfileName] = c.CA.Issuance.Profile
+	} else {
+		for name, profileConfig := range c.CA.Issuance.CertProfiles {
+			_, ok := certProfilesMap[name]
+			if !ok {
+				certProfilesMap[name] = profileConfig
+			} else {
+				cmd.Fail(fmt.Sprintf("Duplicate certificate profile name \"%s\" configured", name))
+			}
+		}
+	}
+
+	_, ok := certProfilesMap[c.CA.Issuance.DefaultCertificateProfileName]
+	if !ok {
+		cmd.Fail(fmt.Sprintf("defaultCertificateProfileName:\"%s\" was configured, but a profile object was not found for that name", c.CA.Issuance.DefaultCertificateProfileName))
+	}
```

Comment:
No need to go to quite this much trouble here -- makeCertProfileMaps will handle all this checking. Instead, I think all we need to do is convert the old config into the new config:
```suggestion
	if len(c.CA.Issuance.CertProfiles) == 0 {
		c.CA.Issuance.CertProfiles[c.CA.Issuance.DefaultCertificateProfileName] = c.CA.Issuance.Profile
	}
```